src/main/java/edu/cornell/kfs/module/ar/document/web/struts/CuContractsGrantsInvoiceDocumentAction.java

```diff
@@ -0,0 +1,58 @@
+package edu.cornell.kfs.module.ar.document.web.struts;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.struts.action.ActionForm;
+import org.apache.struts.action.ActionForward;
+import org.apache.struts.action.ActionMapping;
+import org.kuali.kfs.kns.util.KNSGlobalVariables;
+import org.kuali.kfs.krad.util.ErrorMessage;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.module.ar.document.ContractsGrantsInvoiceDocument;
+import org.kuali.kfs.module.ar.document.web.struts.ContractsGrantsInvoiceDocumentAction;
+import org.kuali.kfs.module.ar.document.web.struts.ContractsGrantsInvoiceDocumentForm;
+import org.kuali.kfs.module.cg.businessobject.Award;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+
+import edu.cornell.kfs.module.cg.businessobject.AwardExtendedAttribute;
+import edu.cornell.kfs.sys.CUKFSKeyConstants;
+
+public class CuContractsGrantsInvoiceDocumentAction extends ContractsGrantsInvoiceDocumentAction {
+    private static final Logger LOG = LogManager.getLogger(CuContractsGrantsInvoiceDocumentAction.class);
+    
+    @Override
+    public ActionForward prorateBill(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) throws Exception {
+        ContractsGrantsInvoiceDocumentForm contractsGrantsInvoiceDocumentForm = (ContractsGrantsInvoiceDocumentForm) form;
+        KualiDecimal budgetTotalAmount = findAwardBUdgetTotal(contractsGrantsInvoiceDocumentForm);
+        
+        if (budgetTotalAmount == null || budgetTotalAmount.isLessEqual(KualiDecimal.ZERO)) {
+            String budetTotalAmountString = budgetTotalAmount != null ? budgetTotalAmount.toString() : "0"; 
+            ErrorMessage errorMessage = new ErrorMessage(CUKFSKeyConstants.ERROR_DOCUMENT_CONTRACT_GRANT_INVOICE_DOCUMENT_PRORATE_NO_AWARD_BUDGET_TOTAL, budetTotalAmountString);
+            KNSGlobalVariables.getMessageList().add(errorMessage);
+            
+            LOG.error("prorateBill, Prorate is not valid as the budgetTotalAmount is " + budgetTotalAmount);
+
+            return mapping.findForward(KFSConstants.MAPPING_BASIC);
+        }
+        
+        return super.prorateBill(mapping, form, request, response);
+    }
+
+    private KualiDecimal findAwardBUdgetTotal(ContractsGrantsInvoiceDocumentForm contractsGrantsInvoiceDocumentForm) {
```

Comment:
Minor typo in method name; should be changed from "...BUdgetTotal" to "...BudgetTotal"

Reply:
ok